packages/ember-intl/addon/services/intl.ts

```diff
@@ -138,6 +140,27 @@ export default class IntlService extends Service {
     return formatDate(intlShape, value, options);
   }
 
+  formatDateTimeRange(
+    from: FormatDateTimeRangeParameters[0] | undefined | null,
+    to: FormatDateTimeRangeParameters[1] | undefined | null,
+    options?: FormatDateTimeRangeParameters[2] & {
+      locale?: string;
+    },
+  ): string {
+    if (
+      from === undefined ||
+      from === null ||
+      to === undefined ||
+      to === null
+    ) {
+      return '';
+    }
```

Comment:
Can you temporarily comment out this code, to check how `formatjs` handles these 3 edge cases?

- Only `from` is `undefined` or `null`.
- Only `to` is `undefined` or `null`.
- Both `from` and `to` are `undefined` or `null`.

Assuming that we want to show an empty string for all 3 cases, let's split the conditional into two for simplicity:

```ts
if (from === undefined || from === null) {
  return '';
}

if (to === undefined || to === null) {
  return '';
}
```

Reply:
| from/to     | null                                | undefined | new Date() | false |
| ---            | ---                                 | ---  |--- |---
| null           | `1/1/1970`                     | (throws Invalid time value error) | `1/1/1970 – 12/23/2024` | `1/1/1970`  |
| undefined  |  (throws Invalid time value error)        | (throws Invalid time value error) | (throws Invalid time value error) | (throws Invalid time value error) |
| new Date() | `12/23/2024 – 1/1/1970` |  (throws Invalid time value error) | `12/23/2024` | `12/23/2024 – 1/1/1970` |
| false           | `1/1/1970`                     | (throws Invalid time value error) | `1/1/19 70 – 12/23/2024` | `1/1/1970` |

In summary:
- `null` and `false` behave just like `0`, which is the beginning of unix epoch "1/1/1970"
- `undefined` is invalid and hard errors

Keep in mind that:
- in ember-intl's service `formatDate`, we are already doing
```ts
if (value === undefined || value === null) {
  return '';
}
```

So, we are forcing `null` being `''`  instead of `1/1/1970`.
I replicated the same conditions to keep things consistent between helpers.

I will do the separate arg validations as you suggested. Please let me know if these findings change something and you would like me to change other validations across the board.

Reply:
Thanks for checking how `formatjs` handles `undefined` and `null`.

For me, at least, it makes more sense to show an empty string by default (versus showing `1/1/1970` or causing a runtime error) when the values aren't well-defined. So let's keep the early exits as is, similarly to what we do for other `format*()` methods.